code/src/App.js

```diff
@@ -1,9 +1,60 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+import ThoughtItem from 'components/ThoughtItem';
+import ThoughtForm from 'components/ThoughtForm';
 
 export const App = () => {
+  const [thoughtItem, setThoughtItem] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [newThought, setNewThought] = useState('');
+
+  useEffect(() => {
+    // eslint-disable-next-line no-use-before-define
+    fetchThoughts();
+  }, []);
+
+  const fetchThoughts = () => {
+    setLoading(true);
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts')
+      .then((res) => res.json())
+      .then((data) => setThoughtItem(data))
+      .catch((error) => console.error(error))
+      .finally(() => setLoading(false));
+  }
+
+  const handleNewThoughtChange = (event) => {
+    setNewThought(event.target.value)
+  }
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        message: newThought
+      })
+    }
+
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts', options)
+      .then((res) => res.json())
+      .then(() => fetchThoughts())
+      .finally(() => setNewThought(''));
+  }
+
   return (
     <div>
-      Find me in src/app.js!
+      <ThoughtForm
+        newThought={newThought}
+        onNewThoughtChange={handleNewThoughtChange}
+        onFormSubmit={onFormSubmit} />
+      <ThoughtItem
+        loading={loading}
+        thoughtItem={thoughtItem}
+        setThoughtItem={setThoughtItem} />
     </div>
   );
 }
```

Comment:
Despite of what you said in the demo, I think just making this work and understand each line, it's already good enough. We will be learning all the time. From what I see here, you managed to make a clean code lines and I understand everything you wrote here. I have a suggestion, I wrote so much comments on my codes for myself to just repeat and relearn everything. That helped me a lot and maybe you can try that as well.